worker/upload.go

```diff
@@ -292,14 +305,27 @@ func (w *worker) uploadMultiPart(ctx context.Context, r io.Reader, bucket, path,
 		return "", fmt.Errorf("couldn't add multi part: %w", err)
 	}
 
-	// if packing was enabled try uploading packed slabs in a separate goroutine
+	// if packing was enabled try uploading packed slabs
 	if up.packing {
-		go w.tryUploadPackedSlabs(up.rs, up.contractSet)
+		// if we reached the buffer size limit, upload one packed slab
+		// synchronously to ensure we block and don't let the buffer grow out of
+		// control
+		if bufferSizeLimitReached {
+			ctx, cancel := context.WithTimeout(ctx, defaultPackedSlabsUploadTimeout)
+			if _, err := w.uploadPackedSlabs(ctx, defaultPackedSlabsLockDuration, up.rs, up.contractSet, defaultPackedSlabsLimit); err != nil {
+				w.logger.Errorf("couldn't upload packed slabs, err: %v", err)
+			}
+			cancel()
+		}
+
+		// make sure there's a goroutine uploading the remainder of the packed slabs
+		go w.threadedUploadPackedSlabs(up.rs, up.contractSet)
```

Comment:
Maybe there is a nice way to deduplicate this code? Maybe by moving this logic into `uploadPackedSlabs`.